src/render/Renderer.cpp

```diff
@@ -654,6 +654,17 @@ void CHyprRenderer::renderWindow(CWindow* pWindow, CMonitor* pMonitor, timespec*
 }
 
 void CHyprRenderer::renderLayer(SLayerSurface* pLayer, CMonitor* pMonitor, timespec* time, bool popups) {
+    static auto PDIMAROUND = CConfigValue<Hyprlang::FLOAT>("decoration:dim_around");
+
+    if (*PDIMAROUND && pLayer->dimAround && !m_bRenderingSnapshot && !popups) {
+        CBox monbox = {0, 0, g_pHyprOpenGL->m_RenderData.pMonitor->vecTransformedSize.x, g_pHyprOpenGL->m_RenderData.pMonitor->vecTransformedSize.y};
+        g_pHyprOpenGL->renderRect(&monbox, CColor(0, 0, 0, *PDIMAROUND * pLayer->alpha.value()));
+
+        if (pLayer->alpha.isBeingAnimated()) {
+            g_pHyprRenderer->damageMonitor(g_pCompositor->getMonitorFromID(pLayer->monitorID));
```

Comment:
you cannot damage while rendering. I'd recommend damaging in an onUpdate callback for the layer alpha avar.

Reply:
Done (At least I'm pretty sure `startAnimation` is the correct place to register the callback)